selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/details/WorkspaceDetailsProjectsTest.java

```diff
@@ -0,0 +1,199 @@
+import org.eclipse.che.selenium.pageobject.dashboard.Dashboard;
+import org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceDetails;
+import org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceProjects;
+import org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceProjectsSamples;
+import org.eclipse.che.selenium.pageobject.dashboard.workspaces.Workspaces;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class WorkspaceDetailsProjectsTest {
+  private static final String PROJECT_NAME = NameGenerator.generate("wsDetails", 4);
+  private static final String PROJECT_FOR_SEARCHING_NAME = NameGenerator.generate("searchWs", 4);
+  private static final String SPRING_SAMPLE_NAME = "web-java-spring";
+  private static final String CONSOLE_SAMPLE_NAME = "console-java-simple";
+  private static final String EXPECTED_SUCCESS_NOTIFICATION = "Success\n" + "Workspace updated.";
+  private static final String CHECKING_CHECKBOXES_PROJECT_NAME = "checkboxesProject";
+  private static final String CHECKING_CHECKBOXES_NEVEST_PROJECT_NAME = "newestProject";
+  private static final String PROJECTS_TAB_NAME = "Projects";
+  private static final String EXPECTED_DELETE_BUTTON_MESSAGE =
+      "Projects cannot be removed in stopped workspace.";
+
+  @Inject private Dashboard dashboard;
+  @Inject private WorkspaceProjects workspaceProjects;
+  @Inject private TestWorkspace testWorkspace;
+  @Inject private TestWorkspace checkingCheckboxesWorkspace;
+  @Inject private Workspaces workspaces;
+  @Inject private WorkspaceDetails workspaceDetails;
+  @Inject private WorkspaceProjectsSamples workspaceProjectsSamples;
+  @Inject private TestWorkspaceServiceClient testWorkspaceServiceClient;
+  @Inject private TestProjectServiceClient testProjectServiceClient;
+  @Inject private DefaultTestUser defaultTestUser;
+  @Inject private SeleniumWebDriverHelper seleniumWebDriverHelper;
+  @Inject private ProjectExplorer projectExplorer;
+
+  @BeforeClass
+  public void setup() throws Exception {
+
+    Path projectPath =
+        Paths.get(getClass().getResource("/projects/default-spring-project").toURI());
+    testProjectServiceClient.importProject(
+        testWorkspace.getId(), projectPath, PROJECT_NAME, MAVEN_SPRING);
+
+    testProjectServiceClient.importProject(
+        testWorkspace.getId(), projectPath, PROJECT_FOR_SEARCHING_NAME, MAVEN_SPRING);
+
+    testProjectServiceClient.importProject(
+        checkingCheckboxesWorkspace.getId(),
+        projectPath,
+        CHECKING_CHECKBOXES_PROJECT_NAME,
+        MAVEN_SPRING);
+
+    testProjectServiceClient.importProject(
+        checkingCheckboxesWorkspace.getId(),
+        projectPath,
+        CHECKING_CHECKBOXES_NEVEST_PROJECT_NAME,
+        MAVEN_SPRING);
+
+    testWorkspaceServiceClient.stop(
+        checkingCheckboxesWorkspace.getName(), defaultTestUser.getName());
+
+    dashboard.open();
```

Comment:
It's hard to understand what we are expecting at the next commands will be executing. Short comment in code with explanation of goal could help.